machine_learning/fp_growth.py

```diff
@@ -0,0 +1,335 @@
+"""
+The FP-Growth (Frequent Pattern Growth) algorithm is a widely used
+data mining technique for discovering frequent itemsets in
+large transaction databases.
+It overcomes some of the limitations of traditional methods like
+Apriori by efficiently constructing the FP-Tree
+
+WIKI: https://athena.ecs.csus.edu/~mei/associationcw/FpGrowth.html
+Examples: https://www.javatpoint.com/fp-growth-algorithm-in-data-mining
+"""
+
+from typing import Optional
+
+
+class TreeNode:
```

Comment:
Please consider making this a https://docs.python.org/3/library/dataclasses.html